src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -409,24 +409,28 @@ def on_request(self, request):
 
         # 'request_data' has been dumped into JSON string in set_json_body() when building HttpRequest in Python SDK.
         # In order to facilitate subsequent parsing, it is converted into a dict first
-        http_request.data = json.loads(request_data)
+        modified_data = json.loads(request_data)
 
-        if http_request.data.get('properties', {}).get('template'):
```

Comment:
Out of curiosity, does this mean that deployment commands never worked if `properties` only contain `templateLink` 🤔?

Reply:
It works today because of the `plug_pipeline` parameter:

https://github.com/Azure/azure-cli/blob/a9a87c7dbc7a50ce93c7f753b814e3991a49328e/src/azure-cli/azure/cli/command_modules/resource/custom.py#L491

It's not very discoverable that this is linked to `JsonCTemplatePolicy`'s special-casing of "templateLink". In my case, I was trying to support a `.bicepparam` file linked directly to a template spec, so I ran into this bug because neither `template_uri` nor `template_spec` were being set.

My guess is that someone else ran into the same bug and added the `plug_pipeline` workaround rather than fixing the root issue.